### PR TITLE
Update OTel dependency to v1.14.0/v0.37.0

### PR DESCRIPTION
**Pull request by zchee**
Update each OTel dependency to v1.14.0/v0.37.0, and format `go.mod`.
Note that upgrade collector, and each contrib package are based v1.14.0/v0.37.0.

Release note:
- https://github.com/open-telemetry/opentelemetry-go/releases/tag/v1.14.0

Also, remove use the deprecated (removed) `go.opentelemetry.io/otel/metric/unit` package, and unified semconv version with `v1.18.0`.

- go.opentelemetry.io/otel@v1.14.0
- go.opentelemetry.io/otel/metric@v0.37.0
- go.opentelemetry.io/otel/metric/instrument@v1.14.0
- go.opentelemetry.io/otel/sdk@v1.14.0
- go.opentelemetry.io/otel/sdk/export/metric@v0.37.0
- go.opentelemetry.io/otel/sdk/metric@v0.37.0
- go.opentelemetry.io/otel/trace@v1.15.0
- go.opentelemetry.io/otel/exporters/prometheus@v0.37.0
- go.opentelemetry.io/contrib/detectors/gcp@v1.15.0
- go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp@v0.40.0
- go.opentelemetry.io/collector@v0.73.0
- go.opentelemetry.io/collector/featuregate@v0.73.0
- go.opentelemetry.io/collector/semconv@v0.73.0
- go.opentelemetry.io/collector/pdata@v1.0.0-rc7
- github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/prometheus@v0.73.0

---

Close: #611

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -179,19 +179,19 @@ for-all-package:
...
+UNSTABLE_OTEL_VERSION=v0.37.0
+  STABLE_CONTRIB_OTEL_VERSION=v1.15.0
 UNSTABLE_CONTRIB_OTEL_VERSION=v0.40.0
-STABLE_CONTRIB_OTEL_VERSION=v1.15.0
-UNSTABLE_COLLECTOR_VERSION=v0.72.0
-STABLE_COLLECTOR_VERSION=v1.0.0-rc6
-UNSTABLE_COLLECTOR_CONTRIB_VERSION=v0.72.0
+  STABLE_COLLECTOR_VERSION=v1.0.0-rc7
+UNSTABLE_COLLECTOR_VERSION=v0.73.0
+UNSTABLE_COLLECTOR_CONTRIB_VERSION=v0.73.0
```
Comment (damemi) [inline Makefile:194]: are the spaces at the start of these lines intentional?

**Review by damemi — APPROVED** [review]
thanks for understanding @zchee, I think this is a good incremental change. just 1 whitespace question. I'll check the branch protection rules. Might just have to override those when I merge, then I'll remove them once I can verify that the 1.20 compatibility tests run in the next PR

**Comment by zchee** [comment]
@damemi PTAL

**Comment by zchee** [comment]
@damemi thanks!

**Comment by damemi** [comment]
@zchee Thanks for your patience!

**Comment by zchee** [comment]
@damemi BTW, are you planned cut new release include this changes?

**Comment by damemi** [comment]
@zchee yeah we can do a release with these libraries then another once the SDK libraries are GA

**Comment by zchee** [comment]
@damemi Ah, I mean, are you planned create "v0.37.0 and v1.13.0" (or etc)? Because upgraded each OTel dependency, also "update go version to 1.20" is worth to new version.

**Comment by zchee** [comment]
@damemi Ah thanks! sorry for misreading cuz I'm not good English :P
I can't wait to cut new release!

**Pull request merged**